### ISSUE TITLE
UNUSUAL lightningd: Unable to estimate opening fees

### ISSUE THREAD

**Issue by GlenCooper** [fiatjaf/lightningd-gjson-rpc#4]
Hi fiatjaf.  Thanks for putting this together.  Was working great for over a week that I was using it, but then suddenly started spewing errors about a day or two ago.  I haven't had time to try to figure out why... figured I should at least raise an issue here.

Please help:

root@dabih-420:~# lightningd
plugin-trustedcoin  initialized plugin v0.2
plugin-trustedcoin tip: 641833
plugin-trustedcoin returning block 641818, 000000000000000000007c8669…, 1504582 bytes
2020-08-02T02:57:43.144Z INFO lightningd: --------------------------------------------------
2020-08-02T02:57:43.144Z INFO lightningd: Server started with public key 02fd6bfd779a590e5d5c12115c8743b12b8e5376f69f0b7977542b40a00f370ecb, alias GREENCHEF (color #02fd6b) and
 lightningd 0.8.2.1
plugin-trustedcoin returning block 641819, 0000000000000000000f791fe5…, 1288447 bytes
plugin-trustedcoin returning block 641820, 00000000000000000000e85cf5…, 1366796 bytes
**plugin-trustedcoin estimatefees error: invalid character '<' looking for beginning of value**
**2020-08-02T02:58:04.865Z UNUSUAL lightningd: Unable to estimate opening fees
2020-08-02T02:58:04.865Z UNUSUAL lightningd: Unable to estimate mutual_close fees
2020-08-02T02:58:04.866Z UNUSUAL lightningd: Unable to estimate unilateral_close fees
2020-08-02T02:58:04.866Z UNUSUAL lightningd: Unable to estimate delayed_to_us fees
2020-08-02T02:58:04.866Z UNUSUAL lightningd: Unable to estimate htlc_resolution fees
2020-08-02T02:58:04.866Z UNUSUAL lightningd: Unable to estimate penalty fees
2020-08-02T02:58:04.867Z UNUSUAL lightningd: Unable to estimate min_acceptable fees
2020-08-02T02:58:04.867Z UNUSUAL lightningd: Unable to estimate max_acceptable fees**
plugin-trustedcoin returning block 641821, 0000000000000000000a5f2738…, 1271929 bytes
**plugin-trustedcoin getblock error: failed to get raw block <!doctype html>**

**Comment by evd0kim** [comment]
I also have messages of `UNUSUAL` type but it seems nothing disrupting my node actually.

**Comment by fiatjaf** [comment]
`estimatefees error: invalid character '<' looking for beginning of value`. This means the esplora hosts (which should be 4 different ones in the latest version -- but maybe that wasn't included in the latest release, so I guess 2 different ones) are returning invalid responses to `/fee-estimates` calls.

`plugin-trustedcoin getblock error: failed to get raw block <!doctype html>`. This means the 2 raw block providers are returning invalid responses for block calls.

Which is odd is why suddenly everybody is returning invalid responses. I got a similar error when DNS stopped working on my VPS a few days ago.

Can you call `curl https://blockstream.info/api/fee-estimates` manually from your terminal and get a valid JSON response?

**Comment by GlenCooper** [comment]
root@acrux-420:~# date
Sun Aug  9 04:51:38 CEST 2020
root@acrux-420:~# curl https://blockstream.info/api/fee-estimates
{"16":24.500999999999998,"9":111.531,"1008":1.248,"5":111.76,"8":111.531,"18":24.500999999999998,"21":24.500999999999998,"504":1.248,"22":24.500999999999998,"17":24.500999999999998,"14":24.500999999999998,"1":142.67,"10":111.531,"12":111.531,"15":24.500999999999998,"19":24.500999999999998,"23":24.500999999999998,"144":2.239,"13":24.507,"3":111.76,"7":111.531,"11":111.531,"24":24.500999999999998,"25":14.159,"6":111.76,"20":24.500999999999998,"4":111.76,"2":142.67}root@acrux-420:~#

**Comment by fiatjaf** [comment]
Does this error keep happening all the time?

**Comment by GlenCooper** [comment]
Not all the time.  I started lightningd in a screen session on this host, and saw the error messages in the screen session when I reattached to it to check on it.

I was able to run "curl https://blockstream.info/api/fee-estimates" a bunch of times manually in succession and get what appeared to be valid JSON results every time.

I'm going to close out this issue.  Thanks again for this, and all that you do
![Screenshot from 2020-08-18 07-13-23](https://user-images.githubusercontent.com/967793/90482194-6356ec00-e122-11ea-9588-ba41c43ed002.png)
